### PR TITLE
Fixes double qdel in wielded

### PR DESCRIPTION
**Pull request by Archanial**
## About The Pull Request

Clean up the code a bit.
Tested on local worked.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

## Changelog
:cl:
fix: fixed double qdel in wielded
/:cl:

### CODE REVIEW

File: code/datums/components/twohanded.dm [hunk 1]
```diff
@@ -154,10 +150,7 @@
 	if(ignore_attack_self)
 		return
 
-	if(wielded)
-		unwield()
-	else
-		wield(user)
+	wielded ? unwield() : wield(user)
```
Comment (PowerfulBacon) [inline code/datums/components/twohanded.dm:153]: this is kind of cursed

File: code/datums/components/twohanded.dm [hunk 2]
```diff
@@ -140,12 +140,8 @@
 
 	if(auto_wield)
 		UnregisterSignal(user, COMSIG_MOB_SWAP_HANDS)
-	if(require_twohands)
-		unwield(TRUE)
-	if(wielded)
+	if(require_twohands || wielded)
 		unwield()
```
Comment (PowerfulBacon) [inline code/datums/components/twohanded.dm:144]: What did the TRUE do and was it important?

**Review by PowerfulBacon — APPROVED** [review]

File: code/datums/components/twohanded.dm (hunk 2 above)
Comment (PowerfulBacon) [reply code/datums/components/twohanded.dm:144]: This changes the behaviour of unwield by making it not display a message when you unwield a twohanded item.

Comment (Archanial) [reply code/datums/components/twohanded.dm:144]: it's true by default

**Pull request merged**